src/gui/newwizard/states/basiccredentialssetupwizardstate.cpp

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) Fabian Müller <fmueller@owncloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#include "basiccredentialssetupwizardstate.h"
+#include "jobs/checkbasicauthjobfactory.h"
+#include "networkjobs/fetchuserinfojobfactory.h"
+
+namespace OCC::Wizard {
+
+BasicCredentialsSetupWizardState::BasicCredentialsSetupWizardState(SetupWizardContext *context)
+    : AbstractSetupWizardState(context)
+{
+    _page = new BasicCredentialsSetupWizardPage(_context->accountBuilder().serverUrl());
+}
+
+void BasicCredentialsSetupWizardState::evaluatePage()
+{
+    auto *basicCredentialsSetupWizardPage = qobject_cast<BasicCredentialsSetupWizardPage *>(_page);
+    OC_ASSERT(basicCredentialsSetupWizardPage != nullptr);
```

Comment:
```suggestion
    Q_ASSERT(basicCredentialsSetupWizardPage != nullptr);
```

Reply:
There is no way how to continue here. The client will crash in the next line. This situation should *never* ever occur.

Reply:
OC_ENFORCE

Reply:
But is there a point in checking it in release builds if its basically impossible to happen?
I mean thats what asserts are for, to validate the code during development and testing.